tests/plaid/test_plaid.py

```diff
@@ -0,0 +1,231 @@
+                'income': {},
+                'expenses': {},
+                'investments': {
+                    'earnings': {},
+                    'deposits': {},
+                    'withdrawals': {}
+                }
+            },
+            'savings': {
+                'general': {},
+                'earnings': {},
+                'cash_flow': {
+                    'deposits': {},
+                    'withdrawals': {}
+                }
+            }
+        }
+
+        if credit_card:
+            credit_card = dict_reverse_cumsum(credit_card, 'amount', 'current')
+            metadata = general(metadata, credit_card, 'credit_card')
+            metadata = late_payment(metadata, credit_card)
+
+        if checking:
+            checking = dict_reverse_cumsum(checking, 'amount', 'current')
+            metadata = general(metadata, checking, 'checking')
+            metadata = income(metadata, checking, 'sub_category', 'payroll')
+            metadata = expenses(metadata, checking, 'sub_category', 'rent')
+            metadata = expenses(metadata, checking, 'sub_category', 'insurance')
+            metadata = expenses(metadata, checking, 'sub_category', 'utilities')
+            metadata = expenses(metadata, checking, 'sub2_category', 'loans and mortgages')
+            metadata = investments(metadata, checking, 'sub2_category', 'financial planning and investments')
+
+        if savings:
+            savings = dict_reverse_cumsum(savings, 'amount', 'current')
+            metadata = general(metadata, savings, 'savings')
+            metadata = cash_flow(metadata, savings, 'category', 'interest')
+            metadata = earnings(metadata, savings, 'sub_category', 'interest earned')
+
+        self.mtd = metadata
+
+
+# -------------------------------------------------------------------------- #
+#                      some state-based UNIT TEST CASES                      #
+#                - test core functions of Plaid algorithm -                  #
+# -------------------------------------------------------------------------- #
+
+
+class TestMetricCredit(unittest.TestCase):
+
+    def setUp(self):
+        ''' import test values (inputs) that will feed app functions '''
+
+        # expected result
+        self.expected = [1, 0.69, 1.0, 0.17, 0.38, 1, 0, 1]
+
+        # inputs
+        self.fb = PlaidFeedback().__dict__['fb']
+        self.prm = PlaidParams().__dict__['prm']
+        self.prd = PlaidParams().__dict__['prd']
+        self.mtd = PlaidMetadata().__dict__['mtd']
+
+    def tearDown(self):
+        ''' reset test values after running tests'''
+
+        self.fb = None
+        self.prm = None
+        self.prd = None
+        self.mtd = None
+
+    def test_credit(self):
+        ''' perform test calling app functions '''
+
+        s, f = plaid_credit_metrics(self.fb, self.prm, self.mtd, self.prd)
+
+        self.assertCountEqual(s, self.expected)
+        self.assertListEqual(s, self.expected)
```

Comment:
That's great and hopefully we can write more tests in the future. Right now we're simply asserting equality between the user's score for each metric and the expected score. Yet, we are not testing for type match, raise errors, raise exception, and function behaviour at a granular level. It would be useful to test for the behaviour of each unit of code (a function) in the future.